x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtIssuer.java

```diff
@@ -94,6 +94,37 @@ record AlgJwkPair(String alg, JWK jwk) {}
         }
     }
 
+    public void rotate(
```

Comment:
This method duplicates most of the code in the constructor. After a certain line:
- Missing update to `this.algorithms`
- HTTPS server contents is updated

It should be possible for a single method to handle a call from constructor or rotate. That would reduce duplication, and avoid a copy-and-paste issues like missing update to `this.algorithms`.